### ISSUE TITLE
[Suggestion] ./scripts/dev_setup.sh, fresh Ubuntu 18.04: doesn't include apt update, fails if not run

### ISSUE THREAD

**Issue by AronVanAmmers** [libra/libra#19]
# Suggestion

When running `./scripts/dev_setup.sh` on a fresh Ubuntu 18.04 VM where `apt update` or `apt-get update` hasn't recently been run, the script fails with:

```
E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
```

When I run `sudo apt update` manually, then run `./scripts/dev_setup.sh` this error doesn't occur.

While it isn't strictly the responsibility of the `dev_setup.sh` script to ensure package info is up to date, the goal is to provide a one-command install, so it would help if it does. I suggest adding a `sudo apt update` after `apt-get` has been detected as the package manager.

## To reproduce

* Establish an Ubuntu / Debian system where the package data is out of date. For example, create a new VM with Vagrant, image `ubuntu/bionic64`
* Run `./scripts/dev_setup.sh`

** Stack trace/error message **
```
E: Failed to fetch http://archive.ubuntu.com/ubuntu/pool/main/b/binutils/binutils-common_2.30-21ubuntu1~18.04_amd64.deb  404  Not Found [IP: 91.189.91.23 80]
E: Failed to fetch http://archive.ubuntu.com/ubuntu/pool/main/b/binutils/libbinutils_2.30-21ubuntu1~18.04_amd64.deb  404  Not Found [IP: 91.189.91.23 80]
E: Failed to fetch http://archive.ubuntu.com/ubuntu/pool/main/b/binutils/binutils-x86-64-linux-gnu_2.30-21ubuntu1~18.04_amd64.deb  404  Not Found [IP: 91.189.91.23 80]
E: Failed to fetch http://archive.ubuntu.com/ubuntu/pool/main/b/binutils/binutils_2.30-21ubuntu1~18.04_amd64.deb  404  Not Found [IP: 91.189.91.23 80]
E: Failed to fetch http://security.ubuntu.com/ubuntu/pool/main/liba/libarchive/libarchive13_3.2.2-3.1ubuntu0.1_amd64.deb  404  Not Found [IP: 91.189.91.23 80]
E: Failed to fetch http://archive.ubuntu.com/ubuntu/pool/main/g/gcc-7/gcc-7-base_7.3.0-27ubuntu1~18.04_amd64.deb  404  Not Found [IP: 91.189.91.23 80]
E: Failed to fetch http://archive.ubuntu.com/ubuntu/pool/main/g/gcc-7/cpp-7_7.3.0-27ubuntu1~18.04_amd64.deb  404  Not Found [IP: 91.189.91.23 80]
E: Failed to fetch http://archive.ubuntu.com/ubuntu/pool/main/g/gcc-defaults/cpp_7.3.0-3ubuntu2.1_amd64.deb  404  Not Found [IP: 91.189.91.23 80]
E: Failed to fetch http://archive.ubuntu.com/ubuntu/pool/main/g/gcc-8/libcc1-0_8.2.0-1ubuntu2~18.04_amd64.deb  404  Not Found [IP: 91.189.91.23 80]
E: Failed to fetch http://archive.ubuntu.com/ubuntu/pool/main/g/gcc-8/libgomp1_8.2.0-1ubuntu2~18.04_amd64.deb  404  Not Found [IP: 91.189.91.23 80]
E: Failed to fetch http://archive.ubuntu.com/ubuntu/pool/main/g/gcc-8/libitm1_8.2.0-1ubuntu2~18.04_amd64.deb  404  Not Found [IP: 91.189.91.23 80]
E: Failed to fetch http://archive.ubuntu.com/ubuntu/pool/main/g/gcc-8/libatomic1_8.2.0-1ubuntu2~18.04_amd64.deb  404  Not Found [IP: 91.189.91.23 80]
E: Failed to fetch http://archive.ubuntu.com/ubuntu/pool/main/g/gcc-7/libasan4_7.3.0-27ubuntu1~18.04_amd64.deb  404  Not Found [IP: 91.189.91.23 80]
E: Failed to fetch http://archive.ubuntu.com/ubuntu/pool/main/g/gcc-8/liblsan0_8.2.0-1ubuntu2~18.04_amd64.deb  404  Not Found [IP: 91.189.91.23 80]
E: Failed to fetch http://archive.ubuntu.com/ubuntu/pool/main/g/gcc-8/libtsan0_8.2.0-1ubuntu2~18.04_amd64.deb  404  Not Found [IP: 91.189.91.23 80]
E: Failed to fetch http://archive.ubuntu.com/ubuntu/pool/main/g/gcc-7/libubsan0_7.3.0-27ubuntu1~18.04_amd64.deb  404  Not Found [IP: 91.189.91.23 80]
E: Failed to fetch http://archive.ubuntu.com/ubuntu/pool/main/g/gcc-7/libcilkrts5_7.3.0-27ubuntu1~18.04_amd64.deb  404  Not Found [IP: 91.189.91.23 80]
E: Failed to fetch http://archive.ubuntu.com/ubuntu/pool/main/g/gcc-8/libmpx2_8.2.0-1ubuntu2~18.04_amd64.deb  404  Not Found [IP: 91.189.91.23 80]
E: Failed to fetch http://archive.ubuntu.com/ubuntu/pool/main/g/gcc-8/libquadmath0_8.2.0-1ubuntu2~18.04_amd64.deb  404  Not Found [IP: 91.189.91.23 80]
E: Failed to fetch http://archive.ubuntu.com/ubuntu/pool/main/g/gcc-7/libgcc-7-dev_7.3.0-27ubuntu1~18.04_amd64.deb  404  Not Found [IP: 91.189.91.23 80]
E: Failed to fetch http://archive.ubuntu.com/ubuntu/pool/main/g/gcc-7/gcc-7_7.3.0-27ubuntu1~18.04_amd64.deb  404  Not Found [IP: 91.189.91.23 80]
E: Failed to fetch http://archive.ubuntu.com/ubuntu/pool/main/g/gcc-defaults/gcc_7.3.0-3ubuntu2.1_amd64.deb  404  Not Found [IP: 91.189.91.23 80]
E: Failed to fetch http://security.ubuntu.com/ubuntu/pool/main/l/linux/linux-libc-dev_4.15.0-42.45_amd64.deb  404  Not Found [IP: 91.189.91.23 80]
E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
```

## Expected Behavior

Script completes without fault

## System information

Libra version: current cloned from github
OS version: Ubuntu 18.04 in Vagrant, box `ubuntu/bionic64`

**Comment by spartanz51** [comment]
You have to update your deposit database before trying to install something, like dependencies..
So of course, make sure to run sudo apt-get update before

**Comment by AronVanAmmers** [comment]
@spartanz51 agreed, the classification "Bug" hence is not appropriate. Renamed to "Suggestion". Unfortunately I can't edit the issue label.

**Comment by jsonsivar** [comment]
This worked for me on Ubuntu 19.04 as well :+1:

**Comment by desthercz** [comment]
**sudo apt update** should be definitelly included in the script in my opinion, there is no reason not to - that will only spawn useless github issues reporting it as error.

**Comment by vsilent** [comment]
For those who don't want to waste time on building docker image for tests, you can get from.
```
trydirect/libra_mint

trydirect/libra_e2e
```

**Comment by nealmcb** [comment]
This appears to be addressed by e58a013, but with a nasty side-effect.
It causes the script to immediately ask for a sudo password.
Very bad form!
This update should happen after the `Welcome to Libra!` message.
(And that welcome should provide better alternatives than "trust us blindly with root or hit C-c now")

**Comment by icarter09** [comment]
@AronVanAmmers do you think this ticket should be closed and then create an 'Enhancement' ticket addressing your suggestion?

**Comment by AronVanAmmers** [comment]
@icarter09 yes happy to. Actually the suggestion has already been implemented in e58a013 so I will close now. Thanks!
